public/pages/Destinations/components/createDestinations/CustomWebhook/QueryParamsEditor.js

```diff
@@ -55,27 +57,32 @@ const handleRenderValueField = (fieldName, index) => (
 const propTypes = {
   type: PropTypes.string.isRequired,
   queryParams: PropTypes.array.isRequired,
+  isEnabled: PropTypes.bool,
 };
 
-const QueryParamsEditor = ({ type, queryParams }) => (
-  <FieldArray
-    name={`${type}.queryParams`}
-    validateOnChange={true}
-    render={arrayHelpers => (
-      <AttributeEditor
-        titleText="Query parameters"
-        onAdd={() => arrayHelpers.push({})}
-        onRemove={index => arrayHelpers.remove(index)}
-        items={queryParams}
-        name={`${type}.queryParams`}
-        addButtonText="Add parameter"
-        removeButtonText="Remove parameter"
-        onRenderKeyField={handleRenderKeyField}
-        onRenderValueField={handleRenderValueField}
-      />
-    )}
-  />
-);
+const QueryParamsEditor = ({ type, queryParams, isEnabled = true }) => {
```

Comment:
Sorry, this might be causing you to flip-flip between implementations, but I think passing down isEnabled makes sense simply because I'd like to also see the addButton disabled (and maybe get @stevensideyliu opinion on if the removeButton should be disabled).

@mihirsoni I think it's still generic enough with that passed down, either the attribute editor is enabled or disabled and inputs/buttons need to reflect that.

Reply:
Sure makes sense. I think so in case of not enabled , we'll disable everything (All inputs , remove button and Add button)

Reply:
Thanks for feedback, as suggested I changed the code to pass down the isEnabled to AttributeEditor and disable Add and Remove buttons if isEnabled is false.